crates/router/src/services/api/client.rs

```diff
@@ -22,7 +22,13 @@ fn get_client_builder(
     proxy_config: &Proxy,
     should_bypass_proxy: bool,
 ) -> CustomResult<reqwest::ClientBuilder, ApiClientError> {
-    let mut client_builder = reqwest::Client::builder().redirect(reqwest::redirect::Policy::none());
+    let mut client_builder = reqwest::Client::builder()
+        .redirect(reqwest::redirect::Policy::none())
+        .pool_idle_timeout(
+            proxy_config
+                .idle_pool_connection_timeout
+                .map(std::time::Duration::from_secs),
+        );
```

Comment:
only apply this function if the value is provided,

if you call it with None, it will set no timeout